src/clj_icat_direct/queries.clj

```diff
@@ -268,15 +271,28 @@
       [(format base "?" "?") user zone]
       (format base (str "'" user "'") (str "'" zone "'")))))
 
+(defn mk-paths-for-uuids
+  [uuids]
+  (sql/format
+   {:with      [[:objid (-> (h/select :object_id :meta_attr_value)
+                            (h/from :r_objt_metamap)
+                            (h/join :r_meta_main [:using :meta_id])
+                            (h/where [:= :meta_attr_name "ipc_UUID"]
+                                     [:= :meta_attr_value (sql/call :any (sql-array "varchar" uuids))]))]]
+    :union-all [(-> (h/select [:meta_attr_value :uuid]
+                              [:coll_name :path])
+                    (h/from :r_coll_main)
+                    (h/join :objid [:= :r_coll_main.coll_id :objid.object_id]))
+                (-> (h/select [:meta_attr_value :uuid]
+                              [(sql/raw "coll_name || '/' || data_name") :path])
+                    (h/modifiers :distinct)
+                    (h/from :r_data_main)
+                    (h/join :r_coll_main [:using :coll_id]
+                            :objid [:= :r_data_main.data_id :objid.object_id]))]}))
```

Comment:
I'm consistently impressed by HoneySQL. I honestly didn't expect it to have support for CTEs.